jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JacocoReportPublisher.java

```diff
@@ -168,6 +168,7 @@ public void process(@Nonnull StepContext context, @Nonnull Element mavenSpyLogsE
             String destFile = destFileElt.getTextContent().trim();
             if (destFile.equals("${jacoco.destFile}")) {
                 destFile = "${project.build.directory}/jacoco.exec";
+                if ("prepare-agent-integration".equals(pluginInvocation.goal)) destFile = "${project.build.directory}/jacoco-it.exec";
```

Comment:
This will set the destFile to the correct default value when this is a prepare-agent-integration